src/aind_behavior_experiment_launcher/behavior_launcher/slims_picker.py

```diff
@@ -0,0 +1,426 @@
+            while subject is None:
+                subject = self.ui_helper.input("Enter subject name: ")
+                try:
+                    self._slims_mouse = self.slims_client.fetch_model(SlimsMouseContent, barcode=subject)
+                except exceptions.SlimsRecordNotFound:
+                    logger.info(f"No Slims mouse with barcode {subject}. Please re-enter.")
+                    subject = None
+            self.launcher.subject = subject
+
+        sessions = self.slims_client.fetch_models(SlimsBehaviorSession, mouse_pk=self.slims_mouse.pk)
+        try:
+            self._slims_session = sessions[-1]
+        except IndexError:  # empty list returned from slims
+            raise ValueError(f"No session found on slims for mouse {subject}.")
+
+        notes = self.ui_helper.prompt_text("Enter notes: ")
+
+        return self.launcher.session_schema_model(
+            experiment="",  # Will be set later
+            root_path=str(self.launcher.data_dir.resolve())
+            if not self.launcher.group_by_subject_log
+            else str(self.launcher.data_dir.resolve() / subject),
+            subject=subject,
+            notes=notes + "\n" + (self.slims_session.notes if self.slims_session.notes else ""),
+            experimenter=experimenter if experimenter is not None else [],
+            commit_hash=self.launcher.repository.head.commit.hexsha,
+            allow_dirty_repo=self.launcher.is_debug_mode or self.launcher.allow_dirty,
+            skip_hardware_validation=self.launcher.skip_hardware_validation,
+            experiment_version="",  # Will be set later
+        )
+
+    def pick_task_logic(self) -> TTaskLogic:
+        """
+        Returns task_logic found as an attachment from session loaded from slims.
+
+        Returns:
+            TTaskLogic: Task logic found as an attachment from session loaded from slims.
+
+        Raises:
+            ValueError: If no valid task logic attachment is found.
+        """
+
+        try:
+            return self.launcher.task_logic_schema
+
+        except ValueError:
+            # check attachments from loaded session
+            attachments = self.slims_client.fetch_attachments(self.slims_session)
+            try:  # get most recently added task_logic
+                response = [
+                    self.slims_client.fetch_attachment_content(attach).json()
+                    for attach in attachments
+                    if attach.name == ByAnimalFiles.TASK_LOGIC.value
+                ][0]
+            except IndexError:  # empty attachment list with loaded session
+                raise ValueError(
+                    "No task_logic model found on with loaded slims session for mouse"
+                    f" {self.launcher.subject}. Please add before continuing."
+                )
+
+            return self.launcher.task_logic_schema_model(**response)
+
+    def write_behavior_session(
+        self,
+        task_logic: TTaskLogic,
```

Comment:
We should also do a TODO here to use curriculum state instead of tasklogic down the road.